src/AuthenticatedAPIClient/tests/AuthenticatedAPIClient.test.jsx

```diff
@@ -155,179 +150,299 @@ describe('getAuthenticatedAPIClient', () => {
-      expectRequestToHaveCsrfToken(mock.history.post[0]);
-      expectRequestToHaveJwtAuth(mock.history.post[0]);
-      expectRequestToHaveCsrfToken(mock.history.post[1]);
-      expectRequestToHaveJwtAuth(mock.history.post[1]);
+  // This test case is unexpected, but occurring in production. See ARCH-948 for
+  // more information on a similar situation that was happening prior to this
+  // refactor in Oct 2019.
+  describe('Unexpected case where token refresh succeeds but the cookie is malformed', () => {
+    beforeEach(() => {
+      setJwtCookieTo('malformed jwt');
+      // The JWT cookie is malformed despite a 200 response.
+      setJwtTokenRefreshResponseTo(200, 'malformed jwt');
     });
-  });
 
-  it('fetches a csrf token from the host in the BASE_URL if the url is a path', () => {
-    return client.post('/path/endpoint')
-      .then(() => {
-        expectSingleCallToJwtTokenRefresh();
-        expectSingleCallToCsrfTokenFetch();
-        expectRequestToHaveCsrfToken(mock.history.post[0]);
-        expectRequestToHaveJwtAuth(mock.history.post[0]);
-        expect(csrfTokensManagerAxiosMock.history.get[0].url)
-          .toEqual(`${global.location.origin}${authConfig.csrfTokenApiPath}`);
+    ['get', 'options', 'post', 'put', 'patch', 'delete'].forEach((method) => {
+      it(`${method.toUpperCase()}: throws an error and redirects`, () => {
+        expect.hasAssertions();
+        return client[method](mockApiEndpointPath).catch(() => {
+          expectSingleCallToJwtTokenRefresh();
+          expectNoCallToCsrfTokenFetch();
+          expect(logError).toHaveBeenCalledWith(
+            'frontend-auth: Error decoding JWT token.',
+            expect.any(Object),
+          );
+          expectLogout();
+        });
       });
+    });
   });
 });
 
-describe('A GET request when the user is logged out', () => {
+describe('User is logged out', () => {
   beforeEach(() => {
+    setJwtCookieTo(null);
     setJwtTokenRefreshResponseTo(401, null);
   });
 
-  it('redirects to login when no token exists and refreshing fails', () => {
-    setJwtCookieTo(null);
-    return client.get(mockApiEndpointPath).then(() => {
-      expectSingleCallToJwtTokenRefresh();
-      expectLogin();
+  ['get', 'options', 'post', 'put', 'patch', 'delete'].forEach((method) => {
+    it(`${method.toUpperCase()}: redirects to login`, () => {
+      return client[method](mockApiEndpointPath).then(() => {
+        expectSingleCallToJwtTokenRefresh();
+        expectLogin();
+      });
     });
   });
+});
+
+beforeEach(() => {
+  setJwtCookieTo(null);
+  setJwtTokenRefreshResponseTo(200, jwtTokens.valid.encoded);
+});
 
-  it('redirects to login when an expired token exists and refreshing fails', () => {
-    setJwtCookieTo(jwtTokens.expired.encoded);
-    return client.get(mockApiEndpointPath).then(() => {
-      expectSingleCallToJwtTokenRefresh();
-      expectLogin();
+describe('Backstop Logging for api requests', () => {
```

Comment:
Maybe: `Info logging for authorization errors from api requests with a valid token.`?  I was confused again about this case until I remembered the difference from the 401 and 403 tests above.

Reply:
Done.